src/oneD/Sim1D.cpp

```diff
@@ -724,6 +724,80 @@ double Sim1D::fixedTemperatureLocation()
     return z_fixed;
 }
 
+void Sim1D::setLeftControlPoint(double temperature)
+{
+    double epsilon = 1e-3; // Precision threshold for being 'equal' to a temperature
+    for (size_t n = 0; n < nDomains(); n++) {
+        Domain1D& d = domain(n);
+
+        // Skip if the domain type doesn't match
+        if (d.domainType() != "axisymmetric-flow") {
+            continue;
+        }
+
+        StFlow* d_axis = dynamic_cast<StFlow*>(&domain(n));
```

Comment:
I'd suggest
```suggestion
        StFlow& d_axis = dynamic_cast<StFlow&>(domain(n));
```
so it will raise an exception if the `dynamic_cast` fails, rather than just returning a null pointer that then crashes the program when it's dereferenced on the next line.

Reply:
I made this change and added a new accessor function for setting the left and right control point protected variables as they are called in these method.